lib/config/presets/internal/replacements.ts

```diff
@@ -38,6 +39,54 @@ export const presets: Record<string, Preset> = {
       },
     ],
   },
+  'emotion-10-to-11': {
+    description:
+      'various unscoped emotion v10 packages became renamed & scoped in v11',
+    packageRules: [
+      {
+        matchCurrentVersion: '>=11.0.0',
+        matchDatasources: ['npm'],
+        matchPackageNames: ['@emotion/core', 'emotion-theming'],
```

Comment:
Those rule can give duplicate dependency, would that be a problem?

Reply:
Oh really? I had hoped renovate would deduplicate the dependencies?

It's a little awkward, because `create-emotion` is replaced by `@emotion/css/create-instance` as an entry point of the new `@emotion/css` package. So by default, even if we deduplicate, the source code would blow up due to the API change. But, that's outside the scope of the replacement and up to consumers to fix on their end via post upgrade tasks.

Reply:
Ah yes, certainly does produce duplicates.

``` diff
"dependencies": {
-    "@emotion/core": "^11.0.0",
-    "emotion": "^11.0.0",
-    "emotion-theming": "^11.0.0",
-    "emotion-server": "^11.0.0",
-    "create-emotion": "^11.0.0",
-    "create-emotion-server": "^11.0.0"
+    "@emotion/react": "^11.8.1",
+    "@emotion/css": "^11.7.1",
+    "@emotion/react": "^11.8.1",
+    "@emotion/server": "^11.4.0",
+    "@emotion/css": "^11.7.1",
+    "@emotion/server": "^11.4.0"
  }
```

Anyone opposed to renovate deduplicating automatically in this situation? (outside the scope of this PR of course).

Reply:
Created #18919 to track this. I am going to try and merge this, and we can follow-up later.